src/Donations/Controllers/DonationRequestController.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace Give\Donations\Controllers;
+
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationRoute;
+use phpDocumentor\Reflection\Types\Boolean;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonationRequestController
+{
+    /**
+     * @var Boolean
+     */
+    private $isAdmin;
+
+    /**
+     * @var Boolean
+     */
+    private $isAnonymousDonationsRedacted;
+
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     */
+    public function getDonation(WP_REST_Request $request)
+    {
+        $donation = Donation::find($request->get_param('id'));
+
+        if ( ! $donation) {
+            return new WP_Error('donation_not_found', __('Donation not found', 'give'), ['status' => 404]);
+        }
+
+        return new WP_REST_Response($this->escDonation($donation));
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonations(WP_REST_Request $request): WP_REST_Response
+    {
+        $anonymousDonations = $request->get_param('anonymousDonations');
+        $this->isAnonymousDonationsRedacted = 'redact' === $anonymousDonations;
```

Comment:
Let's make an actual Enum class for this:
```php
$anonymousMode = new AnonymousDonationMode($request->get_param('anonymousDonations'));

$anonymousMode->isRedacted();
$anonymousMode->isExcluded();
$anonymousMode->isIncluded();
```

This will provide further validation and make things nice and declarative.

Reply:
Resolved in 06c849945a5d83d494ce824490fb727cf46e427a